### ISSUE TITLE
web-vitals script doesn't get injected

### ISSUE THREAD

**Issue by jonthemiller** [cmorten/cypress-web-vitals#1]
# Issue

In pages where the head tag is not exactly `<head>`, the web-vitals script does not get injected and the test fails with a non-specific error.

Setup:

- node version: 14.16.0
- cypress-web-vitals version: 0.3.0

## Details

This line: https://github.com/cmorten/cypress-web-vitals/blob/main/src/visitWithWebVitalsSnippet.js#L6

Does a string replace looking for `"<head>"` in the response. But, the head tag can have attributes/etc after it. In these cases the string replace doesn't work and the script does not get injected, leading to the following error:

```
TypeError
Cannot read properties of undefined (reading 'lcp')
node_modules/cypress-web-vitals/src/waitForVitals.js:10:1
   8 | 
   9 |   for (const key of Object.keys(thresholds)) {
> 10 |     if (!reports[key]) {
     | ^
  11 |       return false;
  12 |     }
  13 |   }
```

Ran into this when using the plugin against a Nuxt app, whose base HTML looks like this:
```html
<!DOCTYPE html>
<html {{ HTML_ATTRS }}>
  <head {{ HEAD_ATTRS }}>
    {{ HEAD }}
  </head>
  <body {{ BODY_ATTRS }}>
    {{ APP }}
  </body>
</html>
```
https://nuxtjs.org/docs/concepts/views/#document-apphtml

## Fixes

Not sure if feasible, but it would be helpful to have the `<head>` search string be configurable. 

Locally, I changed the replace argument to be `/\<head\s*\>/g` to get the plugin running.

**Comment by cmorten** [comment]
Ah good spot @jonthemiller - should be very fixable with a more flexible replace matcher as suggested 🙂

I’m sure very feasible to have the injection point configurable - do you have a specific use-case in mind? I suspect top of the head solves the majority of uses (given want the vitals script to run before everything else to get most accurate metrics) but open to suggestions (and PRs!) 😊

**Comment by jonthemiller** [comment]
Yeah you'd want it first so it gets loaded before anything else. Still trying to work through a way to make the value configurable. Hardcoding a regex seems not great.

**Comment by jonthemiller** [comment]
Opened https://github.com/cmorten/cypress-web-vitals/pull/2 to address. It does use a regex but is simple enough to be clear its intent. It preserves any whitespace/attributes/values on the head tag.

**Comment by cmorten** [comment]
Should be available in `0.4.0`.

Release Notes: https://github.com/cmorten/cypress-web-vitals/releases/tag/0.4.0

**Comment by jonthemiller** [comment]
Thanks!
